omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,107 @@
+import org.openmrs.module.webservices.rest.web.RestTestConstants2_0;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Ignore;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.context.request.WebRequest;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+
+public class PatientAllergyController2_0Test extends MainResourceControllerTest {
+	
+	@Before
+	public void init() throws Exception {
+		executeDataSet(RestTestConstants2_0.ALLERGY_TEST_DATA_XML);
+	}
+	
+	@Override
+	public String getURI() {
+		return "patient/" + RestTestConstants2_0.PATIENT_UUID + "/allergy";
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getUuid()
+	 */
+	@Override
+	public String getUuid() {
+		return RestTestConstants2_0.ALLERGY_ID;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest#getAllCount()
+	 */
+	@Override
+	public long getAllCount() {
+		return 0;
+	}
+	
+	@Test
+	public void deleteUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergy(1);
+		Assert.assertFalse(allergy.isVoided());
+		
+		handle(newDeleteRequest(getURI() + "/" + RestTestConstants2_0.ALLERGY_ID, new Parameter("reason", "unit test")));
+		allergy = Context.getPatientService().getAllergy(1);
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+	
+	@Test
+	@Ignore("REST API does not support delete /patient/patient_uuid/allergy/ to delete all allergies")
+	public void deleteAllAllergies() throws Exception
+	{
+		handle(newDeleteRequest(getURI() + "/", new Parameter("reason", "unit test")));
+		Allergy allergy = Context.getPatientService().getAllergy(1);
+		Assert.assertTrue(allergy.isVoided());
+		Assert.assertEquals("unit test", allergy.getVoidReason());
+	}
+	
+	@Test
+	@Ignore("PatientService calls voidAllergy - never purges")
+	public void purgeUniqueAllergy() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergy(1);
+		Assert.assertFalse(allergy.isVoided());
+		
+		handle(newDeleteRequest(getURI() + "/" + RestTestConstants2_0.ALLERGY_ID, new Parameter("purge", "")));
+		Assert.assertNull(Context.getPatientService().getAllergy(1));
+	}
+	
+	@Test
+	@Ignore("REST API does not support delete /patient/patient_uuid/allergy/ to delete all allergies")
+	public void purgeAllAllergies() throws Exception
+	{
+		handle(newDeleteRequest(getURI() + "/", new Parameter("purge", "")));
+		Assert.assertNull(Context.getPatientService().getAllergy(1));
+	}
+	
+	@Test
+	public void saveAllergy() throws Exception
+	{
+		SimpleObject allergy = deserialize(handle(newGetRequest(getURI() + "/" + RestTestConstants2_0.ALLERGY_ID)));
+		allergy.add("allergyId", "10");
```

Comment:
FWIW this is not supposed to work in REST web services (i.e. setting the primary key of a resource you're going to create.)